pkg/cfn/builder/cluster.go

```diff
@@ -134,7 +135,13 @@ func (c *ClusterResourceSet) newResource(name string, resource gfn.Resource) *gf
 
 func (c *ClusterResourceSet) addResourcesForControlPlane(subnetDetails *subnetDetails) {
 	clusterVPC := &gfneks.Cluster_ResourcesVpcConfig{
-		SecurityGroupIds: gfnt.NewSlice(c.securityGroups...),
+		EndpointPublicAccess:  gfnt.NewBoolean(*c.spec.VPC.ClusterEndpoints.PublicAccess),
```

Comment:
These should be using api.IsEnabled

Reply:
This directly dereferences the pointer because the defaulting logic ensures they are set to a default value. `api.IsEnabled` makes sense when a value can be nil.